File: learning/print.c
#include <stdio.h>
#include <string.h>

int main()
{
	printf("1234567890");
	printf("\n");
	printf("__________");
	printf("\n");
	printf("%6d%-3d", 25, 7);
	printf("\n");
	printf("%8.2f", 123.456);
	printf("\n");
	printf("%-10s%4d", "Hello", 42);
	printf("\n");
	printf("%-9.4f", 1.234567);
	printf("\n");

	return 0;
}

File: practice/rust/yt/src/bin/print.rs
// supress warnings
#![allow(unused)]

// enables struct printing format
#[derive(Debug)]
struct Lang {
    lang: String,
    version: String,
}

fn main() {
    let lang = "rust";
    println!("hello {} {}", lang, lang);
    println!("hello {lang}");

    let x = 2;
    println!("{0} x {0} = {1} ", x, x * x);

    let l = Lang {
        lang: "rust".to_string(),
        version: "1.38".to_string(),
    };
    println!("{:?}", l);
    println!("{:#?}", l);
}
